stardogrest/extended_client.go

```diff
@@ -0,0 +1,20 @@
+package stardogrest
+
+import (
+	"github.com/Azure/go-autorest/autorest"
+)
+
+type ExtendedBaseClient struct {
+	BaseClient
+}
+
+func (client *ExtendedBaseClient) SetConnection(url, username, password string) {
+	client.BaseClient.BaseURI = url
+	client.BaseClient.Authorizer = autorest.NewBasicAuthorizer(username, password)
+}
+
+func NewExtendedBaseClient() *ExtendedBaseClient {
+	return &ExtendedBaseClient{
+		BaseClient: New(),
+	}
+}
```

Comment:
Why not just add the method to the base client?
```suggestion
func NewWithBaseURI(baseURI string) BaseClient {
	return BaseClient{
		Client:  autorest.NewClientWithUserAgent(UserAgent()),
		BaseURI: baseURI,
		Authorizer: autorest.NewBasicAuthorizer(username, password)
	}
}
```
or this:
```go
func (client *BaseClient) SetBaseUriWithBasicAuth(url, username, password string) {
	client.BaseURI = url
	client.Authorizer = autorest.NewBasicAuthorizer(username, password)
}
```
The code is in the same package, so you can add methods as you need besides the generated code.

Reply:
Well we decided to not touch the autogenerated code from the autorest.

Reply:
It's not touching it, you just add to it. Besides, you added the `extended_client.go` file, so you touched it already :)

Reply:
I will see how much test code I have to change before applying your suggestion. I hoped that you already reviewed this part before I started with the tests

Reply:
Moreover the autogenerated files have not been updated (touched) that was the goal. But of course your suggestion to put a separate method for BaseClient in another file is a nice solution.

Reply:
> I hoped that you already reviewed this part before I started with the tests

As I mentioned before: I don't review a PR everytime when there's a commit push. I suggest you use the "request review" of GitHub whenever something's ready for review

Reply:
Well I suggest next time we split the ticket as much as possible so we can avoid huge PRs. It will help to control the development flow better thus avoiding requests for review in the middle of the PR. The problem right now is - a small change requires huge amount of code and test code to be updated.